### PR TITLE
fabric: commit unknown fix

### PR DESCRIPTION
**Pull request by adecaro**

### CODE REVIEW

File: platform/fabric/core/generic/committer.go [hunk 1]
```diff
@@ -125,29 +127,36 @@ func (c *Channel) commitUnknown(txID string, block uint64, indexInBlock int, env
...
-		if err := c.FetchAndStoreEnvelope(txID); err != nil {
+		envelopeRaw, err = c.FetchEnvelope(txID)
+		if err != nil {
 			return errors.WithMessagef(err, "failed getting rwset for tx [%s]", txID)
 		}
 	}
 
 	// shall we commit this unknown envelope
-	if ok, err := c.filterUnknownEnvelope(txID); err != nil || !ok {
+	if ok, err := c.filterUnknownEnvelope(txID, envelopeRaw); err != nil || !ok {
```
Comment (alexandrosfilios) [inline platform/fabric/core/generic/committer.go:147]: It doesn't make a big difference, but this method actually needs a `common.Envelope`, so we do a redundant unmarshaling. Then `StoreEnvelope` needs the byte array, so I would calculate both variables `envelope` and `envelopeRaw` in the if statements above. Then I'd pass `envelope` in filterUnknownEnvelope` and `envelopeRaw` in `StoreEnvelope`.

File: platform/fabric/core/generic/rwset.go [hunk 2]
```diff
@@ -70,3 +70,25 @@ func (c *RWSetLoader) GetRWSetFromETx(txID string) (driver.RWSet, driver.Process
...
+		return nil, nil, errors.Wrapf(err, "failed unpacking envelope [%s]", txID)
+	}
+	logger.Debugf("retrieve rws [%s,%s]", c.Channel, txID)
+
+	rws, err := c.Vault.InspectRWSet(upe.Results)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return rws, upe, nil
```
Comment (alexandrosfilios) [inline platform/fabric/core/generic/rwset.go:93]: If you don't need `upe`, you can remove it from the outputs for now.

**Review by alexandrosfilios — APPROVED** [review]
LGTM modulo comments

**Comment by adecaro** [comment]
@alexandrosfilios , if you are okay, I would address your comments in another PR because I want to change the way we handle the unknown transaction :)

**Comment by alexandrosfilios** [comment]
> @alexandrosfilios , if you are okay, I would address your comments in another PR because I want to change the way we handle the unknown transaction :)

Sure totally fine! :)

**Pull request merged**